starter-code/src/FilterableProductTable.js

```diff
@@ -0,0 +1,53 @@
+// FiltearbleProductTable.js
+
+import React, { Component } from 'react';
+import './FilterableProductTable.css';
+import SearchBar from "./SearchBar";
+import ProductTable from "./ProductTable"
+
+class FilterableProductTable extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      filteredData: this.props.products,
+      showOnUncheckData: []
+    }
+    this.handleData = this.handleData.bind(this);
+    this.showOnStock = this.showOnStock.bind(this);
+  }
+
+  handleData = (event) => {
+    let inputValue = event.target.value.toLowerCase();
+    let filteredData = this.props.products.filter((product) => product.name.toLowerCase().startsWith(inputValue));
+    this.setState({
+      filteredData: filteredData
+    })
+  }
+
+  showOnStock = (event) => {
+    if (event.target.checked) {
+     this.setState({
+       showOnUncheckData: this.state.filteredData.slice()
+     })
+      let onStockData = this.state.filteredData.filter(data => data.stocked)
+      this.setState({
```

Comment:
I feel like there should be four values in your state: `allData`, `filteredData`, `inputValue` and `onStock`.  
You can then have three methods:

`handleData` which checks the `inputValue` and `onStock` properties and calls `setState` after returning data filtered by the above properties.
`handleInput` which sets the `inputData` property and calls `handleData`
`handleCheck` which set the `onStock` property and calls `handleData`

I would say even `allData` is unnecessary since the full data set should always be available in the props object.